tox.ini

```diff
@@ -135,28 +131,11 @@ pass_env =
     {[testenv]pass_env}
     CI
     CI_PACKED_CHARMS
+    JUJU_VERSION
 deps =
-    opensearch-py
-    pytest
-    pytest-asyncio
-    juju==2.9.42.1 # juju 3.0.0 has issues with retrieving action results
-    pytest-operator
+    -r {tox_root}/requirements_juju{env:JUJU_VERSION}.txt
```

Comment:
Do we need a separate file? Why not explicitly specify the juju version here as it was before the change?

Reply:
The libjuju version needs to change with the major Juju version, so for Juju 2 we need to use 2.9.42.2 and for Juju 3 we need to use 3.1.0.1. These version numbers don't line up exactly with libjuju, and we want control over the version regardless, so we need to map Juju 2 to a libjuju version 2, and so on with Juju 3. We also can't have if statements (or any other features of shell scripts, as far as I've found) in tox files, so we can't pick a dependency based on what version of Juju is installed, which is why I'm using this workaround with environment variables. I'm very open to other options if you have them :)